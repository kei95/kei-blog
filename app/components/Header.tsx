export default function Header() {
  return (
    <header className="header-body text-white">
      <a href="/" className="text-lg font-bold">
        Logo
      </a>
      <ul className="hidden md:flex flex-row items-center align-middle gap-4">
        <li>
          <a href="#">Home</a>
        </li>
        <li>
          <a href="#">About</a>
        </li>
        <li>
          <a href="#">Contact</a>
        </li>
      </ul>

      <button className="md:hidden">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-6 w-6"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          strokeWidth="2"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            d="M4 6h16M4 12h16m-7 6h7"
          />
        </svg>
      </button>
    </header>
  );
}
