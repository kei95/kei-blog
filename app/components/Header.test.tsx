import { render, screen } from "@testing-library/react";

import Header, { HEADER_TEST_ID } from "./Header";

describe("<Header />", () => {
  test("render the component - it should have expected elements", () => {
    render(<Header />);
    expect(screen.getByTestId(HEADER_TEST_ID)).toMatchSnapshot();
  });

  test("check each link buttons - they should be tagged to expected section", () => {
    render(<Header />);
    const expectedLinks = [
      { to: "/", name: "Kei Yamashita" },
      { to: "/", name: "Home" },
      { to: "/#about", name: "About" },
      { to: "/#career", name: "Career" },
      { to: "/#posts", name: "Posts" },
    ];

    // Iterate over each links - it should find all expected links with its direction value
    for (let i = 0; i < expectedLinks.length; i++) {
      const { to, name } = expectedLinks[i];
      const linkEl = screen.getByRole("link", { name });

      expect(linkEl).toHaveAttribute("href", `${to}`);
    }
  });
});
