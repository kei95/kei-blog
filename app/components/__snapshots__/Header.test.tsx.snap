// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render the component - it should have expected elements 1`] = `
<header
  class="header-body text-white bg-transparent border-transparent z-50"
  data-testid="HEADER_TEST_ID"
>
  <a
    class="text-lg font-bold font-serif"
    href="/"
  >
    Kei Yamashita
  </a>
  <ul
    class="hidden md:flex flex-row items-center align-middle gap-4"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/#about"
      >
        About
      </a>
    </li>
    <li>
      <a
        href="/#career"
      >
        Career
      </a>
    </li>
    <li>
      <a
        href="/#posts"
      >
        Posts
      </a>
    </li>
  </ul>
  <div
    data-testid="SLIDER_MENU_TEST_ID"
  >
    <button
      class="md:hidden"
      type="button"
    >
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16m-7 6h7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <aside
      class="overflow-x-hidden right-0 duration-300 fixed top-0 h-screen bg-background_darken w-0"
    >
      <div
        class="flex flex-col"
      >
        <button
          type="button"
        >
          Ã—
        </button>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/#about"
        >
          About
        </a>
        <a
          href="/#career"
        >
          Career
        </a>
        <a
          href="/#posts"
        >
          Posts
        </a>
      </div>
    </aside>
  </div>
</header>
`;
